### ISSUE TITLE
The investment bug still exist

### ISSUE THREAD

**Issue by myclan123** [LoneGazebo/Community-Patch-DLL#10339]
_1. Mod Version (X.Y.Z). Current Version: 4.2.0_

4.2.0

_3. List of Other Mods_

None

_4. Describe the Issue_

In the former Version 3.9.1, I described a bug about Investment and the policy Imperium, which should +20 production from Border Expansion. But the production  overflown next turn is less than expected.
https://github.com/LoneGazebo/Community-Patch-DLL/issues/10142

Now I discovered this bug also exist when removing forests. Before it the production was 17.6/31 for invested monument. After +40 production from removing, it became 57.6/31. Adding 13.5 production per turn from the city. The overflow next turn suppose to be around 40.1.
But next turn, what I got was only 13.1.

______________________________________________________________________________________
_5. Save Game From 1 Turn Before (ALWAYS ATTACH THIS IF POSSIBLE)_

[AutoSave_0077 BC-0950.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/12838125/AutoSave_0077.BC-0950.zip)

_8. Steps to reproduce the Issue (Optional)_

Please remove the forest near Alexandria

_9. Screenshots of the Issue (Optional)_
![}83J}BJ(W8YL_3 27@}QUE0](https://github.com/LoneGazebo/Community-Patch-DLL/assets/143814249/35a7eee4-5892-4c21-8ec2-a2aeecb56dba)
![(N$C%F N(004X88~DJ@$X5](https://github.com/LoneGazebo/Community-Patch-DLL/assets/143814249/e4a7b7ee-7766-4f3a-b908-ceec80a425e5)

**Comment by reikken** [comment]
This issue singlehandedly makes the mid game a massive pain, causing me to have to micromanage build queues to avoid wasting large amounts of production

**Comment by myclan123** [comment]
> This issue singlehandedly makes the mid game a massive pain, causing me to have to micromanage build queues to avoid wasting large amounts of production

Absolutely, I need to be very careful not to invest on a building when it will cause the production having a overflow.
